docs/building.md

```diff
@@ -0,0 +1,122 @@
+# Building  
+    
+### Dependencies
+
+Programs used for Building:
+
+* **cmake** 3.4 or greater
+* a C++17 compiler 
+
+This project relies on Magnum Engine. Which needs SDL2.
+
+See Option A: https://doc.magnum.graphics/magnum/getting-started.html#getting-started-setup-subproject
+
+Other dependencies are included as submodules in the 3rdparty folder.
+
+
+## Building on Linux
+To build this project do the following steps:
+1. Install cmake and (lib)**sdl2**(-dev) and git
+ `sudo apt-get install cmake libsdl2-dev git`
+ 
+2. Download the repo & cd into it
+
+Make sure submodules are up to date.
+
+```bash
+# clone with --recurse-submodules
+git clone --recurse-submodules https://github.com/TheOpenSpaceProgram/osp-magnum.git
```

Comment:
Recommend adding --depth=10 and --submodule-depth=10 (make sure I'm remembering the names of these correctly !!)